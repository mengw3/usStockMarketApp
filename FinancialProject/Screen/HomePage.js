import * as React from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  Dimensions,
  ImageBackground,
  FlatList,
} from 'react-native';

import { Card, CardTitle, CardContent, CardAction, CardButton } from 'react-native-material-cards'

var screenWidth = Dimensions.get('window').width;
var screenHeight = Dimensions.get('window').height;

export default class HomeScreen extends React.Component {

  /*
  constructor
  */
  constructor(props){
    super(props);
    this.state ={
        isLoading: true,
        dataSource: []
    }
  }

  /*
  Show the layout of Home Screen.
  */
  render() {

    // data = this.setupData();

    return (
        // console.log(this.state.closePrice),
        console.disableYellowBox = true,
        <View style={{flex: 1}}>
        <ImageBackground source={require('../images/background.jpg')}  style={{width: screenWidth, height: screenHeight}}>
            {/* {this.renderHeader()} */}
            {this.renderList()}
        </ImageBackground>
        </View>
    );
  }

  setupData() {
    fetch('http://127.0.0.1:5000/api/stock_list')
      .then((response) => response.json())
      .then((result) => {
        this.setState({
            isLoading: false,
            dataSource: result,
        });
        return result;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  /*
  Show the layout of Header part.
  */
  renderHeader() {
    return (
      <View style={styles.header}>
          <Text style={styles.headerTitle}>Stock List</Text>
      </View>
    );
  }

  renderList() {
    this.setupData()
    // console.log(this.state.dataSource)
    return (
        <View style={{marginBottom: 180, paddingTop:10}}>
            <FlatList
            data={this.state.dataSource}
            keyExtractor={item => item.stock_symbol}
            renderItem={({item}) => 
                <Card style={{backgroundColor: `rgba(0, 0, 77, 0.2)`}}>
                    <CardTitle
                        title={item.stock_name}
                        subtitle={item.stock_symbol}
                    />
                    <CardContent 
                        textStyle={[styles.contentPrice, {color: item.color}]}
                        text={item.close_price}
                    />
                    <CardAction 
                        separator={true} 
                        inColumn={false}>
                        <CardButton
                        onPress={() => {this.props.navigation.navigate('StockInfo', {
                          symbol: item.stock_symbol,
                          name: item.stock_name,
                          });
                        }}
                        title="Info"
                        color="#62698d"
                        />
                        <CardButton
                        onPress={() => {this.props.navigation.navigate('StockData', {
                          symbol: item.stock_symbol,
                          name: item.stock_name,
                          color: item.color,
                          price: item.close_price,
                          difference: item.difference,
                          });
                        }}
                        title="Chart"
                        color="#62698d"
                        />
                        <CardButton
                        onPress={() => {this.props.navigation.navigate('StockModel', {
                          symbol: item.stock_symbol,
                          name: item.stock_name,
                          price: item.close_price,
                          });
                        }}
                        title="Model"
                        color="#62698d"
                        />
                    </CardAction>
                </Card>
                }
            />
        </View>
    );
  }
}

/*
styles
*/
const styles = StyleSheet.create({
  header: {
    flex: 1,
    alignItems: 'center',
    marginTop: 10,
  },
  headerTitle: {
    flexDirection: 'row',
    marginTop: 50,
    marginBottom: 10,
    color: '#3d3d5c',
    fontSize: 30,
    fontWeight: 'bold',
  },
  contentPrice: {
    flexDirection: 'row',
    marginTop: 10,
    marginBottom: 10,
    fontSize: 30,
    fontWeight: 'bold',
  },
  content: {
    flexDirection: 'row',
    marginBottom: 10,
    color: '#8585ad',
    fontSize: 20,
  },
});