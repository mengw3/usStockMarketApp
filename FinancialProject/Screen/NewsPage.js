import * as React from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  Dimensions,
  ImageBackground,
  FlatList,
  Linking,
} from 'react-native';

import { Card, CardTitle, CardContent, CardAction, CardButton } from 'react-native-material-cards'

var screenWidth = Dimensions.get('window').width;
var screenHeight = Dimensions.get('window').height;

export default class NewsScreen extends React.Component {

  /*
  constructor
  */
  constructor(props){
    super(props);
    this.state ={
        isLoading: true,
        dataSource: ""
    }
  }

  /*
  Show the layout of Home Screen.
  */
  render() {

    // data = this.setupData();

    return (
        // console.log(this.state.dataSource),
        <View style={{flex: 1}}>
        <ImageBackground source={require('../images/background.jpg')}  style={{width: screenWidth, height: screenHeight}}>
          {/* {this.renderHeader()} */}
          {this.renderList()}
        </ImageBackground>
        </View>
    );
  }

  setupData() {
    fetch('http://127.0.0.1:5000/api/financial_news')
      .then((response) => response.json())
      .then((result) => {
        this.setState({
            isLoading: false,
            dataSource: result,
        });
        return result;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  /*
  Show the layout of Header part.
  */
  renderHeader() {
    return (
      <View style={styles.header}>
          <Text style={styles.headerTitle}>News and Reports</Text>
      </View>
    );
  }

  renderList() {
    this.setupData()
    // console.log(this.state.dataSource)
    return (
        <View style={{marginBottom: 180, paddingTop:30}}>
            <Text style={styles.headerTitle}>News and Reports</Text>
            <FlatList
            data={this.state.dataSource}
            keyExtractor={item => item.title}
            renderItem={({item}) => 
                <Card style={{backgroundColor: `rgba(0, 0, 77, 0.2)`}}>
                    <CardTitle
                        title={item.title}
                    />
                    <CardContent 
                        textStyle={styles.contentPrice}
                        text={item.summary}
                    />
                    <CardAction 
                        separator={true} 
                        inColumn={false}>
                        <CardButton
                        onPress={() => Linking.openURL(item.link)}
                        title="More Details"
                        color="#62698d"
                        />
                    </CardAction>
                </Card>
                }
            />
        </View>
    );
  }
}

/*
styles
*/
const styles = StyleSheet.create({
  header: {
    flex: 1,
    alignItems: 'center',
    marginTop: 10,
  },
  headerTitle: {
    flexDirection: 'row',
    marginTop: 30,
    marginBottom: 10,
    color: '#3d3d5c',
    fontSize: 30,
    fontWeight: 'bold',
  },
  contentTitle: {
    flexDirection: 'row',
    marginTop: 10,
    marginBottom: 10,
    color: '#62698d',
    fontSize: 20,
    fontWeight: 'bold',
  },
  content: {
    flexDirection: 'row',
    marginBottom: 10,
    color: '#8585ad',
    fontSize: 20,
  }
});