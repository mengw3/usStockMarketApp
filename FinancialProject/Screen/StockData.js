import * as React from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  Dimensions,
  ImageBackground,
  Button,
} from 'react-native';
import { useRoute } from '@react-navigation/native';
import LineChart from "react-native-responsive-linechart";
import moment from "moment";
import {ButtonGroup} from 'react-native-elements';

var screenWidth = Dimensions.get('window').width;
var screenHeight = Dimensions.get('window').height;

// Wrap and export
export default function(props) {
  const route = useRoute();
  return <DataScreen {...props} route={route} />;
}

class DataScreen extends React.Component {

  /*
  constructor
  */
  constructor(props) {
    super(props);
    this.state = {
        isLoading: true,
        dataSource: [],
        stockData: [],
        stockIndex: [],
        period: "1mo",
        buttonIndex: 1
    };
    this.selectPeriod = this.selectPeriod.bind(this);
  }

  selectPeriod = (newIndex) => {
    if (newIndex == 0) {
      this.setState({
        buttonIndex: newIndex,
        period: "5d"
      })
    }
    if (newIndex == 1) {
      this.setState({
        buttonIndex: newIndex,
        period: "1mo"
      })
    }
    if (newIndex == 2) {
      this.setState({
        buttonIndex: newIndex,
        period: "3mo"
      })
    }
    if (newIndex == 3) {
      this.setState({
        buttonIndex: newIndex,
        period: "6mo"
      })
    }
    if (newIndex == 4) {
      this.setState({
        buttonIndex: newIndex,
        period: "1y"
      })
    }
    if (newIndex == 5) {
      this.setState({
        buttonIndex: newIndex,
        period: "2y"
      })
    }
    if (newIndex == 6) {
      this.setState({
        buttonIndex: newIndex,
        period: "5y"
      })
    }
    if (newIndex == 7) {
      this.setState({
        buttonIndex: newIndex,
        period: "10y"
      })
    }
    if (newIndex == 8) {
      this.setState({
        buttonIndex: newIndex,
        period: "max"
      })
    }
  }

  /*
  Show the layout of Data Screen.
  */
  render() {

    const { route } = this.props;
    
    data = this.setupData(route.params.symbol, this.state.period);

    const buttons = ['5d', '1mo', '3mo', '6mo', '1y', '2y', '5y', '10y', 'max']

    return (
        // console.log(this.state.period),
        // console.log(this.state.stockIndex),
      console.disableYellowBox = true,
      <View style={{flex: 1}}>
        <ImageBackground source={require('FinancialProject/images/black.jpg')}  style={{width: screenWidth, height: screenHeight}}>
            {this.renderHeader(route.params.name, route.params.symbol, route.params.color, route.params.price, route.params.difference)}
            <ButtonGroup
              selectedIndex={this.state.buttonIndex}
              onPress={this.selectPeriod}
              buttons={buttons}
              containerStyle={{height: 40, backgroundColor: 'transparent', marginTop: 20, borderRadius: 10, borderWidth: 2}}
            />
            {this.renderNewChart(route.params.color)}
        </ImageBackground>
      </View>
    );
  }

  setupData(stock_name, stock_period) {
    fetch('http://127.0.0.1:5000/api/stock_data?stock_name=' + stock_name + '&stock_period=' + stock_period)
      .then((response) => response.json())
      .then((result) => {
        const date = result.index.map(element => moment(element).format("M/D"));
        this.setState({
            isLoading: false,
            dataSource: result,
            stockData: result.data,
            stockIndex: date,
        });
        return result;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  /*
  Show the layout of Header part.
  */
  renderHeader(stock_name, stock_symbol, stock_color, stock_price, stock_difference) {
    return (
      <View>
        <Text style = {styles.contentTitle}>{stock_name} ({stock_symbol})</Text>
        <Text style = {styles.contentPrice}>
          <Text style = {{fontSize: 20, fontWeight: 'bold', color: stock_color}}>Current Price: {stock_price}</Text>
          <Text style = {{fontSize: 15, color: stock_color}}>         {stock_difference}</Text>
        </Text>
        <Text style = {styles.content}>recent {this.state.period} data</Text>
      </View>
    );
  }

  renderNewChart(stock_color) {
    const data_temp = this.state.dataSource.data;
    const labels = this.state.stockIndex;
    const config = {
      backgroundColor: "transparent",
      line: {
        visible: true,
        strokeWidth: 5,
        strokeColor: stock_color
      },
      area: {
        visible: false
      },
      tooltip: {
        visible: true,
        labelFormatter: v => v.toFixed(2),
        labelFontSize: 10,
        lineColor: "#777",
        lineWidth: 1,
        circleColor: "#3385ff",
        circleBorderColor: "#3385ff",
        circleBorderWidth: 1,
        boxColor: "#3385ff",
        boxBorderWidth: 1,
        boxBorderColor: "#777",
        boxBorderRadius: 5,
        boxPaddingY: 0,
        boxPaddingX: 0,
        labelColor: "#fff",
        labelFontSize: 20
      },
      grid: {
        visible: false,
        stepSize: 1,
        backgroundColor: "transparent",
      },
      xAxis: {
        visible: false,
        labelFontSize: 5,
        labelColor: "#ffffff"
      },
      yAxis: {
        labelFormatter: v => "$ "+ String(v),
        labelColor: "#ffffff"
      },
      area: {
        visible: false
      },
      insetY: 10,
      insetX: 10
    };
    return (
      <LineChart style={{ flex: 1, alignItems: 'center', marginBottom: 180}} config={config} data={data_temp} xLabels={labels}/>
    );
  }
}

/*
styles
*/
const styles = StyleSheet.create({
  contentTitle: {
    alignItems: 'center',
    flexDirection: 'row',
    marginTop: 10,
    color: '#ffffff',
    fontSize: 20,
    fontWeight: 'bold',
  },
  contentPrice: {
    alignItems: 'center',
    flexDirection: 'row',
    marginBottom: 10,
  },
  content: {
    alignItems: 'center',
    flexDirection: 'row',
    color: '#ffffff',
    fontSize: 20,
  }
});