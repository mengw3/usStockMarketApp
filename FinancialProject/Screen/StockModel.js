import * as React from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  Dimensions,
  ImageBackground,
  ActivityIndicator
} from 'react-native';
import { useRoute } from '@react-navigation/native';
import { LineChart, Grid } from 'react-native-svg-charts'
import moment from "moment";

var screenWidth = Dimensions.get('window').width;
var screenHeight = Dimensions.get('window').height;

// Wrap and export
export default function(props) {
  const route = useRoute();
  return <ModelScreen {...props} route={route} />;
}

class ModelScreen extends React.Component {

  /*
  constructor
  */
  constructor(props) {
    super(props);
    this.state = {
        isLoading_p: true,
        dataSource_p: [],
        prediction: "",
        stockIndex_p: [],
        isLoading_c: true,
        dataSource_c: [],
        stockIndex_c: []
    };
  }

  /*
  Show the layout of Data Screen.
  */
  render() {

    const { route } = this.props;
    
    this.setupData_c(route.params.symbol);

    this.setupData_p(route.params.symbol);

    return (
      console.disableYellowBox = true,
      <View style={{flex: 1}}>
        <ImageBackground source={require('FinancialProject/images/black.jpg')}  style={{width: screenWidth, height: screenHeight}}>
            {this.renderHeader(route.params.name, route.params.symbol)}
            {/* {this.renderNewChart(route.params.symbol)} */}
            {this.renderChart()}
            {this.renderRooter(route.params.price)}
        </ImageBackground>
      </View>
    );
  }

  setupData_p(stock_name) {
    fetch('http://127.0.0.1:5000/api/stock_prediction')
      .then((response) => response.json())
      .then((result) => {
        const date = result[stock_name].index.map(element => moment(element).format("M/D"));
        this.setState({
            isLoading_p: false,
            dataSource_p: result[stock_name],
            prediction: result[stock_name + '-pred'],
            stockIndex_p: date,
        });
        return result;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  setupData_c(stock_name) {
    fetch('http://127.0.0.1:5000/api/stock_original')
      .then((response) => response.json())
      .then((result) => {
        const date = result[stock_name].index.map(element => moment(element).format("M/D"));
        this.setState({
            isLoading_c: false,
            dataSource_c: result[stock_name],
            stockIndex_c: date,
        });
        return result;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  /*
  Show the layout of Header part.
  */
  renderHeader(stock_name, stock_symbol) {
    return (
      <View>
        <Text style = {styles.contentTitle}>{stock_name} ({stock_symbol})</Text>
        <Text style = {[styles.content, {color: '#ffffff'}]}>Prediction Model</Text>
        <Text style = {styles.content_line}>
          <Text style = {{color: '#6666ff'}}>Blue Line</Text>
          <Text style = {{color: '#ffffff'}}> represents Real Data</Text>
        </Text>
        <Text style = {styles.content_line}>
          <Text style = {{color: '#e65c00'}}>Orange Line</Text>
          <Text style = {{color: '#ffffff'}}> represents Prediction Data</Text>
        </Text>
      </View>
    );
  }

  renderNewChart() {
    const data_temp = this.state.dataSource.data;
    const labels = this.state.stockIndex;
    const config = {
      backgroundColor: "transparent",
      line: {
        visible: true,
        strokeWidth: 1,
        strokeColor: "#4747d1"
      },
      area: {
        visible: false
      },
      tooltip: {
        visible: true,
        labelFormatter: v => v.toFixed(2),
        labelFontSize: 10,
        lineColor: "#777",
        lineWidth: 1,
        circleColor: "#4747d1",
        circleBorderColor: "#4747d1",
        circleBorderWidth: 1,
        boxColor: "#4747d1",
        boxBorderWidth: 1,
        boxBorderColor: "#777",
        boxBorderRadius: 5,
        boxPaddingY: 0,
        boxPaddingX: 0,
        labelColor: "#fff",
        labelFontSize: 20
      },
      grid: {
        visible: false,
        stepSize: 1,
        backgroundColor: "transparent",
      },
      xAxis: {
        visible: false,
        labelFontSize: 5,
        labelColor: "#4747d1"
      },
      yAxis: {
        labelFormatter: v => "$ "+ String(v),
        labelColor: "#4747d1"
      },
      area: {
        visible: false
      },
      insetY: 10,
      insetX: 10
    };
    return (
      <LineChart style={{ flex: 1, alignItems: 'center', marginBottom: 180}} config={config} data={data_temp} xLabels={labels}/>
    );
  }

  renderChart() {
    if (this.state.isLoading_p || this.state.isLoading_c) {
      return(
        <View style={{flex: 1, padding: 200}}>
          <ActivityIndicator/>
        </View>
      )
    }

    prediction_data = this.state.dataSource_p.data;
    real_data = this.state.dataSource_c.data;

    const data = [
        {
            data: prediction_data,
            svg: { stroke: '#e65c00' },
        },
        {
            data: real_data,
            svg: { stroke: '#6666ff' },
        },
    ];

    return (
      //  console.log(typeof(prediction_data[0])),
        <View>
          <LineChart
            style={{ height: 500 }}
            data={ data }
            contentInset={{ top: 50, bottom: 50 }}
          >
            <Grid />
          </LineChart>
        </View>
        
    );
  }

  renderRooter(today_price) {
    price_color = "white"
    if (Number(today_price) < Number(this.state.prediction)) {
      price_color = "green"
    }
    if (Number(today_price) > Number(this.state.prediction)) {
      price_color = "red"
    }
    return (
      <View>
        <Text style = {styles.content}>
          <Text style = {{color: '#ffffff'}}>Based on the prediction model, tomorrow's stock price will be: </Text>
          <Text style = {{color: price_color}}>${this.state.prediction}</Text>
          <Text style = {{color: '#ffffff'}}>.</Text>
        </Text>
      </View>
    );
  }

}

/*
styles
*/
const styles = StyleSheet.create({
  contentTitle: {
    alignItems: 'center',
    flexDirection: 'row',
    marginTop: 10,
    marginBottom: 10,
    color: '#ffffff',
    fontSize: 20,
    fontWeight: 'bold',
  },
  content: {
    alignItems: 'center',
    flexDirection: 'row',
    fontSize: 20,
  },
  content_line: {
    alignItems: 'center',
    flexDirection: 'row',
    fontSize: 15,
  }
});