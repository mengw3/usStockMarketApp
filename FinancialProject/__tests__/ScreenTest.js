import 'react-native';
import React from 'react';
import HomePage from '../Screen/HomePage';
import StockInfo from '../Screen/StockInfo';
import StockData from '../Screen/StockData';
import StockModel from '../Screen/StockModel';
import NewsPage from '../Screen/NewsPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders HomePage correctly', async () => {
  const fetch = require('node-fetch');
  global.fetch = fetch
  global.Headers = fetch.Headers;
  const tree = renderer.create(
    <HomePage />
  ).toJSON();
  expect(tree).toMatchSnapshot();
  await new Promise((r) => setTimeout(r, 500));
})

test('renders StockInfo correctly', async () => {
    const fetch = require('node-fetch');
    global.fetch = fetch
    global.Headers = fetch.Headers;
    const tree = renderer.create(
      <StockInfo />
    ).toJSON();
    expect(tree).toMatchSnapshot();
    await new Promise((r) => setTimeout(r, 500));
})

test('renders StockData correctly', async () => {
  const fetch = require('node-fetch');
  global.fetch = fetch
  global.Headers = fetch.Headers;
  const tree = renderer.create(
    <StockData />
  ).toJSON();
  expect(tree).toMatchSnapshot();
  await new Promise((r) => setTimeout(r, 500));
})

test('renders NewsPage correctly', () => {
  const tree = renderer.create(
    <StockModel />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

test('renders NewsPage correctly', () => {
  const tree = renderer.create(
    <NewsPage />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})


// // keys are date and order-of-test based, so just removed them
// const filterKeys = (state) => {
//   if (state.routes) {
//     return {
//       ...state,
//       routes: state.routes.map((route) => {
//         const { key, ...others } = route
//         return filterKeys(others);
//       }),
//     }
//   }
//   return state;
// };
