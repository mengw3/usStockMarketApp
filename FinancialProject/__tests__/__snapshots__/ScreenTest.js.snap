// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomePage correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/background.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "marginBottom": 180,
          "paddingTop": 10,
        }
      }
    >
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;

exports[`renders NewsPage correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/background.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <RCTScrollView
      style={
        Object {
          "flex": 1,
          "marginBottom": 100,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3d3d5c",
                "flexDirection": "row",
                "fontSize": 30,
                "fontWeight": "bold",
                "marginBottom": 10,
                "marginTop": 50,
              }
            }
          >
            News and Reports
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
